SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialWorkerFlags.cpp

```diff
@@ -20,22 +20,58 @@ void USpatialWorkerFlags::ApplyWorkerFlagUpdate(const Worker_FlagUpdateOp& Op)
 	if (Op.value != nullptr)
 	{
 		FString NewValue = FString(UTF8_TO_TCHAR(Op.value));
-		FString& ValueFlag = WorkerFlags.FindOrAdd(NewName);
-		ValueFlag = NewValue;
-		OnWorkerFlagsUpdated.Broadcast(NewName, NewValue);
+		FString& FlagValue = WorkerFlags.FindOrAdd(NewName);
+		FlagValue = NewValue;
+		if (FOnWorkerFlagUpdated* OnWorkerFlagUpdatedPtr = WorkerFlagCallbacks.Find(NewName))
+		{
+			OnWorkerFlagUpdatedPtr->Broadcast(NewName, NewValue);
+		}
+		OnAnyWorkerFlagUpdated.Broadcast(NewName, NewValue);
 	}
 	else
 	{
 		WorkerFlags.Remove(NewName);
 	}
 }
 
-void USpatialWorkerFlags::BindToOnWorkerFlagsUpdated(const FOnWorkerFlagsUpdatedBP& InDelegate)
+void USpatialWorkerFlags::RegisterAnyFlagUpdatedCallback(const FOnAnyWorkerFlagUpdatedBP& InDelegate)
 {
-	OnWorkerFlagsUpdated.Add(InDelegate);
+	OnAnyWorkerFlagUpdated.Add(InDelegate);
 }
 
-void USpatialWorkerFlags::UnbindFromOnWorkerFlagsUpdated(const FOnWorkerFlagsUpdatedBP& InDelegate)
+void USpatialWorkerFlags::RegisterAndInvokeAnyFlagUpdatedCallback(const FOnAnyWorkerFlagUpdatedBP& InDelegate)
 {
-	OnWorkerFlagsUpdated.Remove(InDelegate);
+	RegisterAnyFlagUpdatedCallback(InDelegate);
+	for (const auto& FlagValuePair : WorkerFlags)
+	{
+		InDelegate.Execute(FlagValuePair.Key, FlagValuePair.Value);
+	}
+}
+
+void USpatialWorkerFlags::UnregisterFlagUpdatedCallback(const FString& InFlagName, const FOnWorkerFlagUpdatedBP& InDelegate)
+{
+	if (FOnWorkerFlagUpdated* OnWorkerFlagUpdatedPtr = WorkerFlagCallbacks.Find(InFlagName))
+	{
+		OnWorkerFlagUpdatedPtr->Remove(InDelegate);
+	}
+}
+
+void USpatialWorkerFlags::RegisterFlagUpdatedCallback(const FString& InFlagName, const FOnWorkerFlagUpdatedBP& InDelegate)
+{
+	FOnWorkerFlagUpdated& OnWorkerFlagUpdatedPtr = WorkerFlagCallbacks.FindOrAdd(InFlagName);
+	OnWorkerFlagUpdatedPtr.Add(InDelegate);
```

Comment:
Super nit: Odd to add `Ptr` to the name of a reference, and inconsistent with the same pattern usage in other places.

Reply:
Yeah hold over from when it was a pointer, will fix good catch.